applications/giftless/values.yaml

```diff
@@ -0,0 +1,75 @@
+# Default values for giftless.
+
+# -- Override the base name for resources
+nameOverride: ""
+
+# -- Override the full name for resources (includes the release name)
+fullnameOverride: ""
+
+# -- Resource limits and requests for the giftless frontend pod
+resources: {}
+
+# -- Annotations for the giftless frontend pod
+podAnnotations: {}
+
+# -- Node selector rules for the giftless frontend pod
+nodeSelector: {}
+
+# -- Tolerations for the giftless frontend pod
+tolerations: []
+
+# -- Affinity rules for the giftless frontend pod
+affinity: {}
+
+# The following will be set by parameters injected by Argo CD and should not
+# be set in the individual environment values files.
+image:
+  # -- Giftless image to use
+  repository: docker.io/datopian/giftless
+  # -- Pull policy for the giftless image
+  pullPolicy: "IfNotPresent"
+  # -- Tag of giftless image to use
+  # @default -- The appVersion of the chart
+  tag: ""
+
+ingress:
+  # -- FQDNs of giftless ingresses
+  # @default -- Must be overridden in environment-specific values file
+  hostname:
+    readonly: ""
+    readwrite: ""
+  # -- Additional annotations to add to the ingress
+  annotations: {}
+
+server:
+  # -- Turn on debugging mode
+  debug: false
+  # -- Port for giftless server to listen on
+  port: 5000
+  # -- Number of processes for server
+  processes: 2
+  # -- Number of threads per process
+  threads: 2
+
+# -- Configuration for giftless server
+# @default -- YAML will be used as-is.  cf https://giftless.datopian.com/en/latest/configuration.html
```

Comment:
This is a philosophical thing and I understand the arguments for doing this each way, but I've never been a fan of this way of writing Helm charts for things that we use internally and thus can modify whenever we want. Since this Helm chart is just for us and doesn't need to be general, I would rather make the Helm chart highly opinionated by putting the configuration directly into the config map template and only exposing in `values.yaml` the things that we want to change per environment.

Reply:
I'm not sure that works.  Giftless expects to literally read a YAML file on startup, but we DO need to interpolate environment-specific variables, so it feels like interpolating via values-*.yaml and then running it through toYaml is going to be much easier.

Am I missing something conceptually here?

Reply:
See what Gafaelfawr does:

https://github.com/lsst-sqre/phalanx/blob/main/applications/gafaelfawr/templates/configmap.yaml

Reply:
Per standup, doing it your way.